pkg/resources/fluentbit/tenants.go

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2023 Kube logging authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentbit
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"emperror.dev/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kube-logging/logging-operator/pkg/resources/fluentd"
+	"github.com/kube-logging/logging-operator/pkg/resources/model"
+	"github.com/kube-logging/logging-operator/pkg/resources/syslogng"
+	"github.com/kube-logging/logging-operator/pkg/sdk/logging/api/v1beta1"
+)
+
+type Tenant struct {
+	Name         string
+	AllNamespace bool
+	Namespaces   []string
+}
+
+func FindTenants(ctx context.Context, target metav1.LabelSelector, reader client.Reader) ([]Tenant, error) {
+	var tenants []Tenant
+
+	selector, err := metav1.LabelSelectorAsSelector(&target)
+	if err != nil {
+		return nil, errors.WrapIf(err, "logrouting targetSelector")
+	}
+	listOptions := &client.ListOptions{
+		LabelSelector: selector,
+	}
+	loggingList := &v1beta1.LoggingList{}
+	if err := reader.List(ctx, loggingList, listOptions); err != nil {
+		return nil, errors.WrapIf(err, "listing loggings for targetSelector")
+	}
+	for _, l := range loggingList.Items {
+		l := l
```

Comment:
This seems weird, maybe a leftover from somewhere?

Reply:
this is to avoid the loop variable issue as we are passing a reference of l to a function inside the loop.

also see https://github.com/golang/go/discussions/56010

Reply:
So this is the notorious go for loop issue, thanks!